odometry_node:
  ros__parameters:
    # General
    odom_frame: "odom"
    base_frame: "base_link"
    
    # KISS-ICP odometry parameters
    kiss_slam:
      odometry:
          preprocessing:
              max_range: 50.0
              min_range: 0.0
              deskew: true
          registration:
              max_num_iterations: 500
              convergence_criterion: 0.0001
              max_num_threads: 0
          adaptive_threshold:
              fixed_threshold: -1.0  # -1.0 represents null
              initial_threshold: 2.0
              min_motion_th: 0.1

      local_mapper:
          voxel_size: 0.2
          splitting_distance: 3.0
